### ISSUE TITLE
Workflow should enable users who belong to groups to take workflow actions

### ISSUE THREAD

**Issue by mjgiarlo** [projecthydra-labs/hyrax#227]
<a href="https://github.com/jcoyne"><img src="https://avatars.githubusercontent.com/u/92044?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [jcoyne](https://github.com/jcoyne)**
_Friday Sep 23, 2016 at 15:19 GMT_
_Originally opened as https://github.com/projecthydra/curation_concerns/issues/1032_

----

If I assign a role (e.g. "approvers") to a group, (e.g. "librarians"). The a member of a group should be able to make an approval even if their user does not directly have the "approvers" role.

On the admin page. The user should see things that their group gives them access to act upon.

**Comment by mjgiarlo** [comment]
@vantuyls Do you want to slot this into MD milestone 2 or 3, or leave in its current milestone?

**Comment by julesies** [comment]
Right now groups are assigned to roles in a collection or admin set. Manager groups can review works. Is this story calling for more functionality, another role?

**Comment by no-reply** [comment]
i'm backlogging this because i can't figure out what functionality it's talking about. the groups/roles terminology is so confused throughout the stack that i have no idea whether this is talking about something that's implemented, or something else.

**Comment by rjkati** [comment]
@jlhardes I'm not sure what needs to be tested here. I think this story needs more clarification.

**Comment by jlhardes** [comment]
@no-reply and @julesies are right that groups/roles terminology is kind of hard to parse through between what was originally reported and how things are functioning now. From what I can see on nurax-dev, there are default groups of "public" and "registered" that can have different types of access to works. For collections, you can type in anything you want to create a new group with manager/depositor/viewer access to a collection and "registered" seems to always be available as a group choice as well. I don't see how a user is added to a newly-created group in this collection case, however.

The access roles of manager, depositor, and viewer on collections and access abilities of view/download or edit on works can be assigned to users or to groups but the group assignment portion seems to be unclear, particularly on collections because you can create a group on the spot to add and there doesn't seem to be a way to add users to that group.

There might be something here within the collections case for group creation and user assignment to groups that could be clarified or might need to be fixed. I think #4686 proposes code changes to address this group confusion. Does that seem relevant, @no-reply?

For the second part of the original report, a user logged in to the admin dashboard will see their works and their collections along with works deposited by others that they can manage and collections created by others that they have access to (either deposit, manage, or view). This seems like it is accomplishing the goal of logged in users being able to see things they have access to act upon, regardless of the type of access.

**Comment by jlhardes** [comment]
#2937 (Verification of groups not happening for adding participants/sharing) is closed but also discusses that a user can add any group name to a collection whether or not the group exists. The discussion around this issue ended with deciding that this specific problem would not be fixed but only because there was supposed to be something done elsewhere to rework groups and authentication. I closed this issue in January but the groups problem reported is actually still a problem and maybe 2937 should be reopened since it is related to this issue and #1150 as well.
